rllib/core/rl_module/torch/torch_rl_module.py

```diff
@@ -73,7 +93,23 @@ def get_state(
         inference_only: bool = False,
         **kwargs,
     ) -> StateDict:
-        return convert_to_numpy(self.state_dict())
+        state_dict = self.state_dict()
+        # Filter out `inference_only` keys from the state dict if `inference_only` and
+        # this RLModule is NOT `inference_only` (but does implement the
+        # InferenceOnlyAPI).
+        if (
+            inference_only
+            and not self.config.inference_only
+            and isinstance(self, InferenceOnlyAPI)
```

Comment:
I like this form of testing.

Reply:
Yeah, me, too. This is what Learners should do: Check, whether the given RLModules are actually compatible with them (or expose certain known APIs to work with).
This way, we can eliminate the need to have algo-specific RLModule classes, freeing the user to think about these beforehand and making it easier to switch between algos later.